src/main/java/edu/byu/cs/util/RepoUrlValidator.java

```diff
@@ -1,20 +1,27 @@
 package edu.byu.cs.util;
 
 import edu.byu.cs.autograder.GradingException;
+import edu.byu.cs.autograder.git.GitHelper;
 import org.eclipse.jgit.annotations.Nullable;
 
+import java.io.File;
+import java.util.UUID;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class RepoUrlValidator {
 
     public static boolean isValid(@Nullable String repoUrl) {
+        File cloningDir = new File("./tmp" + UUID.randomUUID());
         try {
-            clean(repoUrl);
+            GitHelper.fetchRepoFromUrl(repoUrl, cloningDir);
```

Comment:
Close. We actually want to perform _both_ checks here to ensure the URL matches all requirements.

Reply:
So do I return the clean check as well?
And, should the fetch use the return of clean or should clean be used by itself just as a check for exceptions?